docs: document type field in ipfs.get output (#3424)
ipfs/js-ipfs: docs/core-api/FILES.md

const cid = 'QmQ2r6iMNpky5f1m4cnm3Yqw8VSvjuKpTcK1X7dBR1LkJF'

for await (const file of ipfs.get(cid)) {
  console.log(file.type, file.path)

  if (!file.content) continue;

  const content = []

  for await (const chunk of file.content) {
    content.push(chunk)
  }

  console.log(content)
}
```

> When invoking this method via the HTTP API client, the response arrives as a stream in the tar format containing the entire contents of the file or directory tree.

Reviewer: the user doesnt care about stream being a tar, maybe its best to not mention tar here to avoid confusion

Author: Good shout, I've updated the text

> When invoking this method via the HTTP API client, the response arrives as a stream in the tar format containing the entire contents of the file or directory tree.

Author: ```suggestion
When invoking this method via the HTTP API client, the response arrives as a stream containing either the entire contents of the file (if the passed [CID][] resolves to a file) or recursive directory tree and all files contained therein (if the passed [CID][] resolves to a directory).
```

> If you are iterating over a directory, in order to proceeed to the next entry in the tar, you must consume the `content` field of the current entry if it is present.

Reviewer: ```suggestion
If you are iterating over a directory, in order to proceed to the next entry in the tar, you must consume the `content` field of the current entry if it is present.
```

> If you are iterating over a directory, in order to proceed to the next entry in the tar, you must consume the `content` field of the current entry if it is present.

Reviewer: ```suggestion
If you are iterating over a directory, in order to proceed to the next entry in the stream, you must consume the `content` field of the current entry if it is present.
```
